addons/purchase_mrp/tests/test_purchase_mrp_flow.py

```diff
@@ -1021,3 +1021,24 @@ def test_valuation_with_backorder(self):
         svl = po.picking_ids[0].move_ids.stock_valuation_layer_ids
         self.assertEqual(svl[0].unit_cost, 67.5)
         self.assertEqual(svl[1].unit_cost, 135000000)
+
+    def test_total_cost_share_rounded_to_precision(self):
+        kit, compo01, compo02 = self.env['product.product'].create([{
+            'name': name,
+            'standard_price': price,
+            'type': 'product',
+        } for name, price in [('Kit', 0), ('Compo 01', 10), ('Compo 02', 20)]])
+
+        self.env['mrp.bom'].create({
+            'product_tmpl_id': kit.product_tmpl_id.id,
+            'type': 'phantom',
+            'bom_line_ids': [(0, 0, {
+                'product_id': compo01.id,
+                'product_qty': 1,
+                'cost_share': 99.99,
+            }), (0, 0, {
+                'product_id': compo02.id,
+                'product_qty': 1,
+                'cost_share': 0.01,
+            })],
+        })
```

Comment:
Can you add at least one assert ?
Like the cost of the kit ?

Reply:
I'm not sure what to assert here. I asserted the `kit.standard_price`, but it's the same as the one I used to create the kit.
I'm not sure how the `cost_share` percentages should affect it. I tried calling the `button_bom_cost` method before asserting the cost, but I noticed that it doesn't account for those percentages either.

Reply:
Sorry for the confusion, When you asked me to assert the cost of the kit, I assumed that it should be influenced by the cost share of its components.

It didn’t seem logical to me to just create a kit product with a `standard_price` and then verify that same price, as I’m usually verifying things that involve some kind of adjustment or update.

In the UI, there's a `button_bom_cost` that calculates the product cost from the BOM, so I thought I might need to use that in this context.